lib/chef/provisioning/transport/ssh.rb

```diff
@@ -85,6 +99,20 @@ def execute(command, execute_options = {})
           end
 
           channel.wait
+
+          @remote_forwards.each do |forward_info|
+              # -R flag to openssh client allows optional :remote_host and
+              # requires the other values so let's do that too.
+              remote_host = forward_info.fetch(:remote_host, 'localhost')
+              remote_port = forward_info.fetch(:remote_port)
+              local_host = forward_info.fetch(:local_host)
+              local_port = forward_info.fetch(:local_port)
+
+              session.forward.cancel_remote(remote_port, remote_host)
+              session.loop { session.forward.active_remotes.include?([remote_port, remote_host]) }
```

Comment:
Hmm, I think the whole thing is inside `with_execute_timeout` so does that handle your concern?